From 0f97dd68a365a2b9eee193776a8a33c20a27afe2 Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Mon, 27 Apr 2020 16:25:36 +0100
Subject: [PATCH] Revert "Merge pull request #17699 from
 fuzzard/cmake_addonsregex"

This reverts commit 9fab130c387245f4a7b9ab29642a29282db34d13, reversing
changes made to 7af85a0fbcfddd3516ea26a8e27e48d9d4eddc73.
---
 cmake/addons/CMakeLists.txt | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/cmake/addons/CMakeLists.txt b/cmake/addons/CMakeLists.txt
index 3dccc014b2..7b217b85bd 100644
--- a/cmake/addons/CMakeLists.txt
+++ b/cmake/addons/CMakeLists.txt
@@ -256,14 +256,9 @@ foreach(addon ${addons})
     if(idx GREATER -1 OR "${ADDONS_TO_BUILD}" STREQUAL "all")
       set(ADDON_FOUND TRUE)
     # Maybe we have a regex
-    else()
-      foreach(ADDONLISTITEM ${ADDONS_TO_BUILD})
-        if(id MATCHES "${ADDONLISTITEM}")
-          message(STATUS "Pattern ${ADDONLISTITEM} matches ${id}, building addon")
-          set(ADDON_FOUND TRUE)
-          break()
-        endif()
-      endforeach()
+    elseif(id MATCHES "${ADDONS_TO_BUILD}")
+      message(STATUS "Pattern ${ADDONS_TO_BUILD} matches ${id}, building addon")
+      set(ADDON_FOUND TRUE)
     endif()
 
     if(ADDON_FOUND)
-- 
2.16.4

